### PR TITLE
Rework Command System

### PR DESCRIPTION
**Pull request by grimofender**
- [x]  Simple command framework
- [ ] Implement Moderator commands seen in other games including: `kick [player]`, `ban [player]`, and `spec [player]`. Can be used by both Custom Owners, and Moderators.
- [ ] Implement Casual commands such as, `start`, `team`, `thugs`, etc...
- [ ] Implement Developer commands for easy testing. `hax`, `shop`,  etc...

### CODE REVIEW

File: src/game-service/entities/player.js [hunk 1]
```diff
@@ -1992,10 +1992,104 @@ Player.onConnect = function(socket, cognitoSub, name, team, partyId){
 	});
 }//End Player.onConnect
 
+const PERM_NONE = 0;
+const PERM_OWNER = 1;
+const PERM_MODERATOR = 2;
+const PERM_DEVELOPER = 3;
+function getPerm(cognitoSub) {
+	let ret = PERM_NONE;
+	if (cognitoSub == createdByCognitoSub)
```
Comment (treatmet) [inline src/game-service/entities/player.js:2001]: createdByCognitoSub is undefined is it not?

Comment (treatmet) [reply src/game-service/entities/player.js:2001]: I could add a global variable that gets set when custom server is created (and reset to "" when the server goes back to ranked). All the custom server settings are global variables in config.js, and the createdByCognitoSub could be one of them.

File: src/game-service/entities/player.js [hunk 2]
```diff
@@ -1992,10 +1992,104 @@ Player.onConnect = function(socket, cognitoSub, name, team, partyId){
...
+command.list = [];
+function getIDFromName(name) {
+	getPlayerList().forEach(player => {
+		if (String(player.name).startsWith(name))
+			return player.id;
+	})
+	return -1;
+}
+
+// TODO: recreate all the commands -.-
```
Comment (treatmet) [inline src/game-service/entities/player.js:2031]: Are you gonna do this, or shall I?

File: src/game-service/entities/player.js (hunk 1 above)
Comment (treatmet) [reply src/game-service/entities/player.js:2001]: By the way, cognitoSub is an Amazon Cognito attribute. Basically a unique key tied to your authenticated user.

**Comment by treatmet** [comment]
Just had a few comments. This is not ready to merge until you finish converting the commands.

Also, I'm not super familiar with working with Forks in GitHub, so I'm not sure how to easily pull and test this locally before merging. I think it would be easier to have you work directly in this repository on a dev branch. I think I made you a contributer. Are you not able to create branches in the main SocketShot repo? I am going to make a dev branch separate from Master, and you can PR to that branch in the future.

I'll DM you in Discord and we can discuss all this. Would probably be easier haha.

**Comment by grimofender** [comment]
> Also, I'm not super familiar with working with Forks in GitHub, so I'm not sure how to easily pull and test this locally before merging. I think it would be easier to have you work directly in this repository on a dev branch. I think I made you a contributer. Are you not able to create branches in the main SocketShot repo? I am going to make a dev branch separate from Master, and you can PR to that branch in the future.

I'll merge it into a new commands branch and open a new pull request to merge it into master

**Pull request merged**